Mage.Sets/src/mage/cards/m/MeteorSwarm.java

```diff
@@ -0,0 +1,51 @@
+package mage.cards.m;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.common.DamageMultiEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.game.Game;
+import mage.target.common.TargetCreatureOrPlaneswalkerAmount;
+import mage.target.targetadjustment.TargetAdjuster;
+
+import java.util.UUID;
+
+/**
+ * @author Raphael-Schulz
+ */
+public final class MeteorSwarm extends CardImpl {
+
+    public MeteorSwarm(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{X}{R}{R}{R}");
+
+        // Meteor Swarm deals 8 damage divided as you choose among X target creatures and/or planeswalkers.
+        this.getSpellAbility().addEffect(
+                new DamageMultiEffect(8).
+                        setText("Meteor Swarm deals 8 damage divided as you choose among X target creatures and/or planeswalkers.")
+        );
+        this.getSpellAbility().addTarget(new TargetCreatureOrPlaneswalkerAmount(8));
+        this.getSpellAbility().setTargetAdjuster(MeteorSwarmAdjuster.instance);
+    }
+
+    private MeteorSwarm(final MeteorSwarm card) {
+        super(card);
+    }
+
+    @Override
+    public MeteorSwarm copy() {
+        return new MeteorSwarm(this);
+    }
+}
+
+enum MeteorSwarmAdjuster implements TargetAdjuster {
+    instance;
+
+    @Override
+    public void adjustTargets(Ability ability, Game game) {
+        ability.getTargets().clear();
+        TargetCreatureOrPlaneswalkerAmount targetCreatureOrPlaneswalkerAmount = new TargetCreatureOrPlaneswalkerAmount(8);
+        targetCreatureOrPlaneswalkerAmount.setMaxNumberOfTargets(ability.getManaCostsToPay().getX());
+        ability.addTarget(targetCreatureOrPlaneswalkerAmount);
```

Comment:
does this work right when X is 0?

Reply:
You are right, it is not working correctly in that case 🤔 
I'm gonna look a bit deeper into that!

Reply:
you can just not add a target at all in that case

Reply:
Nice, yes that works perfectly fine thank you!

I just thought about the rule of the card again and I think that not only the maxNumberOfTargets but the minNumberOfTargets should be equal to the value of X mana paid since the card does not say 'among **up to** X target creatures and/or planeswalkers' but it says 'among X target creatures and/or planeswalkers'.

So this ruling does not apply here I guess:
        // 107.1c If a rule or ability instructs a player to choose “any number,” that player may choose
        // any positive number or zero, unless something (such as damage or counters) is being divided
        // or distributed among “any number” of players and/or objects. In that case, a nonzero number
        // of players and/or objects must be chosen if possible.

Do you think this is correct?
If I have seen that right this is the first card that let's a player divide a fixed number of damage to a flexible number of creature/planeswalkers so I am not 100% sure 😄

Reply:
I have updated the Pull Request accordingly.